lib/core/include/irods/irods_random.hpp

```diff
@@ -1,9 +1,20 @@
 #ifndef IRODS_RANDOM_HPP
 #define IRODS_RANDOM_HPP
 
+#include <string>
+
 namespace irods {
     void getRandomBytes( void * buf, int bytes );
 
+    /// \brief Generate a string of size \p _length consisting of randomly generated alphanumeric characters.
+    ///
+    /// \param[in] _length Desired length of the generated string.
+    ///
+    /// \throws irods::exception If \p _length exceeds max_size or on allocation failure.
```

Comment:
What is "max_size"?
Do we have a variable that defines it?

Reply:
Meant to update this for context: https://en.cppreference.com/w/cpp/string/basic_string/max_size

Reply:
Ah I see. Seems tricky to word that without mentioning `std::string` as an implementation detail.

Reply:
I just mentioned the negative length. Given the smaller type (`std::int16_t`) it is not possible to exceed `max_size` for all the implementations I'm aware of, and allocation failures are not worth mentioning because I imagine other issues will appear first. Please start a new conversation for wording concerns